keyboards/adpenrose/obi/obi.c

```diff
@@ -0,0 +1,31 @@
+// Copyright 2022 Arturo Avila (@ADPenrose)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "obi.h"
+
+/* Encoder */
+#ifdef ENCODER_ENABLE
+bool encoder_update_kb(uint8_t index, bool clockwise) {
+    if (!encoder_update_user(index, clockwise)) { 
+        return false; 
+    }
+/* The switch case allows for different encoder mappings on different layers, "default" map gets applied for all unspecified layers */
```

Comment:
TBH, you want the encoder map feature then. 

https://docs.qmk.fm/#/feature_encoders?id=encoder-map

And yes this works with VIA.
However, encoder map should only be enabled at the keymap level.